src/main.rs

```diff
@@ -104,4 +145,6 @@ async fn shutdown_signal() {
     }
 
     println!("signal received, starting graceful shutdown");
+    // Force shutdown if graceful shutdown takes longer than 10s
+    handle.graceful_shutdown(Some(Duration::from_secs(10)));
```

Comment:
10 seconds could be a bit short for some reductions to complete. How about making it configurable as graceful_timeout with a default of 60s?